File: python/ex011.py
# Faça um programa que leia a largura e altura de uma parede, calcule a área e a quantidade de tinta necessário para pintá-lo
# sabendo que cada litro de tinta, pinta uma área de 2m²
def read_float(prompt):
    while True:
        entry = input(prompt)
        try:
            return float(entry)
        except ValueError:
            print("Type a valid number.")

width = read_float("Largura da parede: ")
height = read_float("Altura da parede: ")

area = width * height
tint = area / 2

print(f"Sua parede tem a dimensão de {width:.2f}x{height:.2f} e sua área é de {area:.2f}m2")
print(f"Para pintar essa parede, você precisa de {tint:.2f}l de tinta.")

File: rust/ex011.rs
use std::io;

fn main() {
    let width = read_float("Largura da parede: ");
    let height = read_float("Altura da parede: ");

    let area = width * height;
    let tint = area / 2.0;

    println!("Sua parede tem a dimensão de {width}x{height} e sua área é de {area}m2");
    println!("Para pintar essa parede, você precisa de {tint}l de tinta.");
}

fn read_float(prompt: &str) -> f32 {
    loop {
        let mut entry = String::new();

        println!("{prompt}");

        io::stdin()
            .read_line(&mut entry)
            .expect("Error reading line.");

        match entry.trim().parse::<f32>() {
            Ok(float_number) => return float_number,
            Err(_) => {
                println!("Type a valid number.")
            }
        }
    }
}
